packages/repeater/src/request-runner/protocols/HttpRequestRunner.ts

```diff
@@ -0,0 +1,244 @@
+    try {
+      if (this.options.headers) {
+        options.setHeaders(this.options.headers);
+      }
+
+      this.logger?.debug(
+        'Executing HTTP request with following params: %j',
+        options
+      );
+
+      const response = await this.request(options);
+
+      return new Response({
+        protocol: this.protocol,
+        statusCode: response.statusCode,
+        headers: response.headers,
+        body: response.body
+      });
+    } catch (err) {
+      return this.handleRequestError(err, options);
+    }
+  }
+
+  private handleRequestError(err: any, options: Request): Response {
+    if (err.response) {
+      const { response } = err;
+
+      return new Response({
+        protocol: this.protocol,
+        statusCode: response.statusCode,
+        headers: response.headers,
+        body: response.body
+      });
+    }
+
+    const message = err.cause?.message ?? err.message;
+    const errorCode = err.cause?.code ?? err.error?.syscall ?? err.name;
+
+    this.logger?.error(
+      'Error executing request: "%s %s HTTP/1.1"',
+      options.method,
+      options.url
+    );
+    this.logger?.error('Cause: %s', message);
+
+    return new Response({
+      protocol: this.protocol,
+      message,
+      errorCode
+    });
+  }
+
+  private async request(options: Request): Promise<IncomingResponse> {
+    let responseTruncatePromise: Promise<Buffer | undefined>;
+
+    const res = await request({
+      agent: this.getRequestAgent(options),
+      body: options.body,
+      followRedirect: false,
+      gzip: true,
+      method: options.method,
+      resolveWithFullResponse: true,
+      strictSSL: false,
+      rejectUnauthorized: false,
+      timeout: this.options.timeout,
+      url: options.url
+    })
+      .on('request', (req: OutgoingMessage) => this.setHeaders(req, options))
+      .on(
+        'response',
+        (response: IncomingResponse) =>
+          (responseTruncatePromise = this.truncateResponse(response))
+      );
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const body = await responseTruncatePromise!;
+
+    if (body) {
+      res.body = body.toString();
+      res.headers['content-length'] = String(body.byteLength);
+    }
+
+    return res;
+  }
+
+  private getRequestAgent(options: Request) {
+    return (
+      this.proxy ??
+      (options.url.startsWith('https') ? this.httpsAgent : this.httpAgent)
+    );
+  }
+
+  private async truncateResponse(
+    res: IncomingResponse
+  ): Promise<Buffer | undefined> {
+    if (res.statusCode === 204 || res.method === 'HEAD') {
+      return;
+    }
+
+    const type = this.parseContentType(res);
+    const maxBodySize = this.maxContentLength * 1024;
+
+    const requiresTruncating = !this.options.whitelistMimes?.some(
+      (mime: string) => type.startsWith(mime)
+    );
+
+    const body = await this.parseBody(res, { maxBodySize, requiresTruncating });
+
+    return body;
```

Comment:
You can remove the redundant local variable

```suggestion
    return this.parseBody(res, { maxBodySize, requiresTruncating });
```